src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java

```diff
@@ -1,6 +1,89 @@
 package com.vinylteam.vinyl.dao.jdbc;
 
 import com.vinylteam.vinyl.dao.UserDao;
+import com.vinylteam.vinyl.dao.jdbc.mapper.UserRowMapper;
+import com.vinylteam.vinyl.entity.User;
+import org.postgresql.ds.PGSimpleDataSource;
+import org.postgresql.util.PSQLException;
+
+import java.sql.*;
 
 public class JdbcUserDao implements UserDao {
+
+    private final PGSimpleDataSource dataSource;
+    private final String COUNT_ALL = "SELECT COUNT(*) FROM public.users";
+    private final String FIND_BY_EMAIL = "SELECT * FROM public.users" +
+            " WHERE email=?";
+    private final String INSERT = "INSERT INTO public.users" +
+            " (\"email\", \"password\", \"salt\", \"iterations\", \"role\")" +
+            " VALUES (?, ?, ?, ?, ?)";
+
+    public JdbcUserDao(PGSimpleDataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    public boolean add(User user) {
+
+        boolean isAdded = false;
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement insertStatement = connection.prepareStatement(INSERT)) {
+            insertStatement.setString(1, user.getEmail());
+            insertStatement.setString(2, user.getPassword());
+            insertStatement.setString(3, user.getSalt());
+            insertStatement.setInt(4, user.getIterations());
+            insertStatement.setString(5, user.getRole().toString());
+
+            insertStatement.executeUpdate();
+            isAdded = true;
+        } catch (PSQLException e) {
+            e.printStackTrace();
```

Comment:
:(

Reply:
Вместо e.printStackTrace(); - throw new RuntimeException("....", e)

Reply:
I have constraints in DB that prohibit adding record if same email already exists, role isn't USER or ADMIN, and email you are trying to add is NULL.
In those cases user doesn't get added, we catch PSQLException and return "false" to indicate that user wasn't added. To avoid "programming on exceptions" PSQLE is caught but not thrown, and I need to know exact error that occurred during adding.
So printStackTrace is there until I implement logger.